File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    phone_book.sort()
    for (let i = 0; i < phone_book.length - 1; i++){
        let idx = phone_book[i].length
        if (phone_book[i] === phone_book[i+1].substring(0, idx)) {
            return false;
        }
    }
    return true
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.HashSet;

class Solution {
    public boolean solution(String[] phone_book) {
        HashSet<String> phoneSet = new HashSet<>();
        
        for (String phone : phone_book) {
            phoneSet.add(phone);
        }

        for (String phone : phone_book) {
            for (int i = 1; i < phone.length(); i++) {
                if (phoneSet.contains(phone.substring(0, i))) {
                    return false;
                }
            }
        }
        
        return true;
    }
}
